test/module/irohad/ametsuchi/ametsuchi_test.cpp

```diff
@@ -632,6 +632,46 @@ TEST_F(AmetsuchiTest, TestingStorageWhenInsertBlock) {
   ASSERT_TRUE(wrapper.validate());
 }
 
+/**
+ * @given created storage
+ * @when commit block
+ * @then committed block is emitted to observable
+ */
+TEST_F(AmetsuchiTest, TestingStorageWhenCommitBlock) {
+  auto log = logger::testLog("TestStorage");
+  ASSERT_TRUE(storage);
+
+  auto expected_block = getBlock();
+
+  // create test subscriber to check if committed block is as expected
+  auto wrapper = make_test_subscriber<CallExact>(storage->on_commit(), 1);
+  wrapper.subscribe([&expected_block](const auto &block) {
+    ASSERT_EQ(*block, expected_block);
+  });
+
+  log->info("Try commit the block");
+
+  std::unique_ptr<MutableStorage> mutable_storage;
```

Comment:
I know that we have framework::expected. But it does not work for mutable storage, because its copies unique ptr in that case

Reply:
Maybe check if it's possible to modify the methods and move the value?